### PR TITLE
chore(onboarding): Update go onboarding to include profiling

### PR DESCRIPTION
**Pull request by Zylphrex**
Go supports profiling. Include it in the onboarding along with the product selection.

### CODE REVIEW

File: static/app/gettingStartedDocs/go/gin.tsx [hunk 1]
```diff
@@ -31,14 +31,26 @@ import (
...
-  Dsn: "${params.dsn}",
+  Dsn: "${params.dsn}",${
+    params.isPerformanceSelected
+      ? `
   EnableTracing: true,
   // Set TracesSampleRate to 1.0 to capture 100%
   // of transactions for performance monitoring.
   // We recommend adjusting this value in production,
-  TracesSampleRate: 1.0,
+  TracesSampleRate: 1.0,`
```
Comment (Zylphrex) [inline static/app/gettingStartedDocs/go/gin.tsx:41]: A few of these go platforms have `EnableTracing: true` as well as `TracesSampleRate: 1.0`. Do we need both? @cleptric do you know?

Comment (cleptric) [reply static/app/gettingStartedDocs/go/gin.tsx:41]: Yes, to enable performance with an empty sample rate to only sample on incoming requests. Go‘s null values required this.

**Comment by cleptric** [comment]
Given the bad state in terms of overhead Go profiling has, we might hold off on this. Not sure though.

**Comment by Zylphrex** [comment]
> Given the bad state in terms of overhead Go profiling has, we might hold off on this. Not sure though.

Are you saying we should not include the profiling onboarding for golang in app for the time being? I think that's fine, it just won't be available in the profiling onboarding either

**Comment by Zylphrex** [comment]
closing in favour of #68750

**Pull request closed without merging**